bin/functions/workload_functions.sh

```diff
@@ -199,7 +199,7 @@ function run_spark_job() {
     export_withlog SPARKBENCH_PROPERTIES_FILES
 
     YARN_OPTS=""
-    if [[ "$SPARK_MASTER" == yarn-* ]]; then
+    if [[ "$SPARK_MASTER" == yarn-* ]] || [[ "$SPARK_MASTER" == yarn ]]; then
```

Comment:
Seems we can just remove the "-" to make it to `if [[ "$SPARK_MASTER" == yarn* ]]` ?

Reply:
It's only a workaround. I think it's better to specifc the value to yarn-client, yarn-cluster, yarn and output error and quit if not these values. Since the mismatch will ignore the rest of parameters all together and will result in unexpected behaviors for users.

Reply:
ok. Merge this first.